metadata-ingestion/tests/integration/nifi/test_nifi.py

```diff
@@ -43,7 +44,11 @@ def test_nifi_ingest(docker_compose_runner, pytestconfig, tmp_path, mock_time):
         )
 
         # Wait for nifi to execute all processors
-        time.sleep(120)
+        url = "http://localhost:9080/nifi-api/flow/process-groups/root"
+        for i in range(12):
+            if requests.get(url).status_code == 200:
+                break
+            time.sleep(10)
```

Comment:
The wait is not only for the nifi to start up but also for nifi workflow to execute and process all events. The current check only verifies if nifi is up. To check whether workflow has run completely and it has generated necessary provenance events, we'll need to call provenance api here and probably check the count of events has reached expected count.

Reply:
Ok, I made a change here, lmk what you think. I'm not exactly sure what the API is saying but this seems like what changes between when the api endpoint first opens and its final state. Would prefer to avoid the full 2 minute wait just to keep things as quick as possible, if possible.